docs/azure/sdk/authentication/create-token-credentials-from-configuration.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Create token credentials from configuration
+description: This article describes how to create Microsoft Entra token credentials from configuration files.
+ms.topic: how-to
+ms.custom: devx-track-dotnet, engagement-fy23
+ms.date: 1/16/2024
+---
+
+# Create Microsoft Entra credential types using configuration files
+
+The `Microsoft.Extensions.Azure` library supports creating different <xref:Azure.Core.TokenCredential?displayProperty=fullName> types from values defined in _appsettings.json_ and other configuration files. This article describes the support for different `TokenCredential` types and how to configure the required values for each type.
+
+## Support for Azure credentials through configuration
+
+The [`Microsoft.Extensions.Azure`](https://www.nuget.org/packages/Microsoft.Extensions.Azure) library can automatically provide Azure service clients with a `TokenCredential` class by searching _appsettings.json_ or other configuration files for credential values using the default `IConfiguration` service for .NET. This approach allows developers to explicitly set credential values across different environments through configuration rather than through app code directly.
+
+The following credential types are supported via configuration:
+
+* [ClientCertificateCredential](#create-a-clientcertificatecredential-type)
```

Comment:
additionallyAllowedTenantsList applies to several of these. I think all except for WorkloadIdentityCredential and ManagedIdentityCredential.

Reply:
Added a section for this.